### PR TITLE
feat: FLP Configuration inquirer

### PR DESCRIPTION
**Pull request by nikmace**
Feat for #2565.
- Adds new inquirer module for Fiori Launchpad Config generator.
- Integrates prompting into `create` package for command `add inbound-navigation`

### CODE REVIEW

File: packages/flp-config-inquirer/src/prompts/utils.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @returns {Platform} An object containing the platform's name and technical identifier.
+ */
+export function getPlatform(): Platform {
+    if (process.mainModule?.filename.includes('yo') || process.stdin.isTTY) {
+        return PLATFORMS.CLI;
+    } else {
+        return isAppStudio() ? PLATFORMS.SBAS : PLATFORMS.VSCODE;
+    }
+}
```
Comment (nikmace) [inline packages/flp-config-inquirer/src/prompts/utils.ts:34]: Maybe this method should be extracted to `environment-check` package?

Comment (cianmSAP) [reply packages/flp-config-inquirer/src/prompts/utils.ts:34]: Please check `getHostEnvironment` in `@sap-ux/fiori-generator-shared`

Comment (nikmace) [reply packages/flp-config-inquirer/src/prompts/utils.ts]: Changed in [1080bf8](https://github.com/SAP/open-ux-tools/pull/2580/commits/1080bf8852c6eedf94422dc2d0479d2642c704e6)

File: packages/flp-config-inquirer/src/prompts/validators.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * Validates that text input does not have zero length and optionally is less than the specified maximum length.
+ * Returns an end user message if validation fails.
+ *
+ * @param input the text input to validate
+ * @param inputName the name of the input as seen by the user
+ * @param maxLength optional, the maximum length of text to allow
+ * @param allowedCharacters optional, define a list of special characters that should be allowed in the input field
+ * @returns true, if all validation checks pass or a message explaining the validation failure
+ */
+export function validateText(
```
[the remark this answers is not in the document]
Comment (nikmace) [reply packages/flp-config-inquirer/src/prompts/validators.ts]: Extracted in [7ec19f4](https://github.com/SAP/open-ux-tools/pull/2580/commits/7ec19f40f16c566719b24afa58b353dcf9bab998)

**Review by cianmSAP — CHANGES_REQUESTED** [review]
All looks good except the validators used to have more specific validation messages for each prompt.
For example, `Semantic Object input is required` vs `Input cannot be empty.` for every prompt.
Can you implement this?

**Review by devinea — DISMISSED** [review]
New package setup for `@sap-ux/flp-config-inquirer` looks good. 
changeset ✅

**Review by cianmSAP — DISMISSED** [review]
New module looks good
I've tested locally and working as expected

**Comment by nikmace** [comment]
There is some problem with builds.. the tests for `@sap-ux/cf-deploy-config-writer:test` fail and that behavior is across many PRs. Not sure what to do.

Also, could someone from @SAP/ux-tools-create review please?

**Review by devinea — DISMISSED** [review]
re-approving

**Review by heimwege — COMMENTED** [review]
Just one single remark: I'm missing navigation config documentation in [create readme](https://github.com/SAP/open-ux-tools/tree/feat/2565/flp-inquirer/packages/create#sap-ux-add) It seems to have been missing the whole time but now that we are adjusting it could you please add a chapter on that one?

**Comment by nikmace** [comment]
> Just one single remark: I'm missing navigation config documentation in [create readme](https://github.com/SAP/open-ux-tools/tree/feat/2565/flp-inquirer/packages/create#sap-ux-add) It seems to have been missing the whole time but now that we are adjusting it could you please add a chapter on that one?

Added in [e1ec16a](https://github.com/SAP/open-ux-tools/pull/2580/commits/e1ec16abc53f88e39d74593d1162231096a2903b)

**Review by heimwege — APPROVED** [review]
- changeset ok
- test coverage very good
- did NOT test locally

**Review by cianmSAP — APPROVED** [review]
Re-approving

**Review by devinea — APPROVED** [review]

**Pull request merged**